frontend/lib/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.tsx

```diff
@@ -158,6 +171,146 @@ export class ArrowVegaLiteChart extends PureComponent<PropsWithHeight, State> {
+        selectors.push(...this.getSelectorsFromChart(spec[type][chart]))
+      }
+    }
+    return selectors
+  }
+
+  public getSelectors(spec: any): string[] {
+    const selectors: string[] = []
+    selectors.push(...this.getSelectorsFromChart(spec))
+    selectors.push(...this.getSelectorsFromCombinedChart(spec, "hconcat"))
+    selectors.push(...this.getSelectorsFromCombinedChart(spec, "vconcat"))
+    return selectors
+  }
+
+  public generateSpec = (): any => {
+    const { element: el, theme } = this.props
+    const spec = JSON.parse(el.spec)
+
+    const { useContainerWidth } = el
+    if (el.vegaLiteTheme === "streamlit") {
+      spec.config = applyStreamlitTheme(spec.config, theme)
+    } else if (spec.usermeta?.embedOptions?.theme === "streamlit") {
+      spec.config = applyStreamlitTheme(spec.config, theme)
+      // Remove the theme from the usermeta so it doesn't get picked up by vega embed.
+      spec.usermeta.embedOptions.theme = undefined
+    } else {
+      // Apply minor theming improvements to work better with Streamlit
+      spec.config = applyThemeDefaults(spec.config, theme)
+    }
+
+    const storedValue = this.props.widgetMgr.getJsonValue(this.props.element)
+    if (storedValue !== undefined) {
+      const selectors = this.getSelectorsFromChart(spec)
+      const parsedStoredValue = JSON.parse(storedValue)
+      if (parsedStoredValue.select) {
+        selectors.forEach(selector => {
+          spec.params.forEach((param: any) => {
+            if (param.name && param.name === selector) {
+              if (param.select.type && param.select.type === "point") {
+                const values = parsedStoredValue.select[selector].vlPoint.or
+                param.select.fields = Object.keys(values[0])
+                param.value = values
+              } else if (param.select.type === "interval") {
+                const values = parsedStoredValue.select[selector]
+                param.value = values
+              }
+            }
+          })
+        })
+      }
+    }
+
+    if (this.props.height) {
+      // fullscreen
+      spec.width = this.props.width
+      spec.height = this.props.height
+    } else if (useContainerWidth) {
+      spec.width = this.props.width
+    }
+
+    if (!spec.padding) {
+      spec.padding = {}
+    }
+
+    if (spec.padding.bottom == null) {
+      spec.padding.bottom = BOTTOM_PADDING
+    }
+
+    if (spec.datasets) {
+      throw new Error("Datasets should not be passed as part of the spec")
+    }
+
+    if (el.isSelectEnabled) {
+      // attempt to add encodings from chart to selection parameter in order to get point interval
+      // This is to ensure that we can consistently recreate state in fullscreen / non fullscreen mode
+      // https://github.com/altair-viz/altair/issues/3285#issuecomment-1858860696
+      if ("params" in spec) {
+        if ("encoding" in spec) {
+          spec.params.forEach((param: any) => {
```

Comment:
this code is relatively unoptimized but gets fixed here: https://github.com/streamlit/streamlit/pull/8304/files#diff-92f0e57b8f2d70900bd5b548ed97c7fb579f2880d48faaea8b7cfac004d479c0